checks/kubernetes_resource.go

```diff
@@ -50,7 +50,7 @@ func (c *KubernetesResourceChecker) Type() string {
 
 func (c *KubernetesResourceChecker) Run(ctx *context.Context) pkg.Results {
 	var results pkg.Results
-	for _, conf := range ctx.Canary.Spec.KubernetesResource {
+	for _, conf := range ctx.Canary.DeepCopy().Spec.KubernetesResource {
```

Comment:
@moshloop I think we should probably make `context.Canary` a private var and always return DeepCopy with a `Canary()` public method.